src/utilities/paths_utils.tsx

```diff
@@ -1,5 +1,9 @@
 import { isUndefined, toPath } from 'lodash'
 
+const pathRegex = /^.{1,64}$/ //TODO: decide on what regex to use.
```

Comment:
`pathToId` produces an HTML id value. See https://www.tutorialspoint.com/What-are-valid-values-for-the-id-attribute-in-HTML for the requirements for id values.

For this function, I think we just need one idRegex for valid id values. 
The logic could be to calculate the id as before and then test it against the idRegex before returning it.

`path` specifically should additionally be validated in the `upsertDoc` function that is inside of `StoreProvider`.
Inside `upsertDoc`, I suggest validating `pathStr` by calling toPath(pathStr) and then checking that all the resulting path components match `/^(0)|([1-9][0-9]{0,4})|([a-z][a-z0-9_]{0,63}(?<!_))$/`. In other words, each path component must ether be a natural number up to 5 digits or a valid snake-case variable name that does not start or end with an underscore. This regex exposes a serious need for unit testing.  At this point, please add a TODO comment to unit test this regex.